apps/src/lib/util/StatsigReporter.js

```diff
@@ -0,0 +1,112 @@
+import logToCloud from '@cdo/apps/logToCloud';
+import {
+  getEnvironment,
+  isProductionEnvironment,
+  isDevelopmentEnvironment,
+} from '../../utils';
+import Statsig from 'statsig-js';
+
+// A flag that can be toggled to send events regardless of environment
+const ALWAYS_SEND = false;
+
+class StatsigReporter {
+  constructor() {
+    const api_element = document.querySelector(
+      'script[data-statsig-api-client-key]'
+    );
+    const api_key = api_element ? api_element.dataset.statsigApiClientKey : '';
+    const managed_test_environment_element = document.querySelector(
+      'script[data-managed-test-server]'
+    );
+    const managed_test_environment = managed_test_environment_element
+      ? managed_test_environment_element.dataset.managedTestServer === 'true'
+      : false;
+    this.local_mode = !(isProductionEnvironment() || managed_test_environment);
+    const options = {
+      environment: {tier: getEnvironment()},
+      localMode: this.local_mode,
+    };
+    if (this.shouldPutRecord(ALWAYS_SEND)) {
+      Statsig.initialize(api_key, options);
+    }
+  }
+
+  // Utilizes Statsig's function for updating a user once we've recognized a sign in
+  async setUserProperties(userId, userType, enabledExperiments) {
+    const formattedUserId = this.formatUserId(userId);
+    const user = {
+      userID: formattedUserId,
+      custom: {
+        type: userType,
+        experiments: enabledExperiments,
+      },
+    };
+    if (!this.shouldPutRecord(ALWAYS_SEND)) {
+      this.log(
+        `User properties: userId: ${formattedUserId}, userType: ${userType}, signInState: ${!!userId}`
+      );
+    }
+    await Statsig.updateUser(user);
+  }
+
+  sendEvent(eventName, payload) {
```

Comment:
Statsig `sendEvent` expects an event name, event value, and a metadata object (optional) ([see here](https://docs.statsig.com/client/jsClientSDK#logging-an-event)). I believe by implementing this like you have it will result in event metrics not formatting correctly in the Statsig dashboard.

Reply:
I am not sure how to make this match our current analytics reporting, as we've formatted all our client side reporting to send an event name and an optional block of additional params. Can we chat through this for a few minutes?

Reply:
We were able to resolve offline- I've left a comment in the code explaining the approach we landed on!